src/model.cpp

```diff
@@ -589,7 +587,10 @@ std::vector<realtype> const& Model::getInitialStates() const {
 }
 
 void Model::setInitialStates(const std::vector<realtype> &x0) {
-    if(x0.size() != (unsigned) nx && x0.size() != 0)
+    if(nx_solver != nx_rdata)
+        throw AmiException("Custom initial conditions are not supported whith conservation laws enabled");
+    // TODO: remove this exception and compute total_cl according to provided x_rdata
```

Comment:
Create issue?

Reply:
This is actually not too difficult to implement now. would probably require some additional tests (which could be added to model presimulation)